src/test/java/com/imgix/test/TestAll.java

```diff
@@ -220,6 +219,80 @@ public void testEncodeDecode() {
 		assertEquals(URLHelper.encodeURIComponent(URLHelper.decodeURIComponent(encodedUrl)), encodedUrl);
 	}
 
+	@Test
+	public void testDomainShardingConstructorDeprecated1() {
+		Class[] classes = new Class[] {
+				String[].class, boolean.class, String.class,
+				URLBuilder.ShardStrategy.class, boolean.class
+		};
+		assertTrue(hasDeprecationAnnotationConstructor(classes));
+	}
+
+	@Test
+	public void testDomainShardingConstructorDeprecated2() {
+		Class[] classes = new Class[] {
+				String[].class
+		};
+		assertTrue(hasDeprecationAnnotationConstructor(classes));
+	}
+
+	@Test
+	public void testDomainShardingConstructorDeprecated3() {
+		Class[] classes = new Class[] {
+				String[].class, boolean.class
+		};
+		assertTrue(hasDeprecationAnnotationConstructor(classes));
+	}
+
+	@Test
+	public void testDomainShardingConstructorDeprecated4() {
+		Class[] classes = new Class[] {
+				String[].class, boolean.class, String.class
+		};
+		assertTrue(hasDeprecationAnnotationConstructor(classes));
+	}
+
+	@Test
+	public void testSetShardStrategyDeprecated() {
+		assertTrue(hasDeprecationAnnotationMethod("setShardStratgy", URLBuilder.ShardStrategy.class));
+	}
+
+	private boolean hasDeprecationAnnotationConstructor(Class[] classes) {
+		Annotation[] annotations = new Annotation[0];
+
+		try {
+			annotations = URLBuilder.class.getConstructor(classes).getAnnotations();
```

Comment:
technically I think it might be possible to just assert `.isAnnotationPresent(Deprecated.class)` here and save yourself a loop later

Reply:
oof, yea that's way easier 😅